src/app/app.component.ts

```diff
@@ -102,10 +91,8 @@ export class AppComponent {
   };
 
   toggleModal = () => {
-    const modalElement: HTMLAbdsModalElement | null = document.querySelector('#modal-1');
-    const currentState = modalElement?.open;
-
-    modalElement?.setAttribute('open', `${!currentState}`);
+    this.isModalOpen = !this.isModalOpen;
+    this.renderer.setAttribute(this.modal.nativeElement, 'open', `${this.isModalOpen ? true : null}`);
```

Comment:
I see what you mean now... we might want to talk about this one as it gives vanillajs vibes 🧐

Reply:
Yea, I did a lot of research to find this solution. This [document](https://ultimatecourses.com/blog/element-refs-in-angular-templates) helped finally get it working if you want to review it.

The issue with modal is that without a reference to it and its attributes from the app, the state gets out of sync easily. Modal uses `open` to display it. You can add a toggle button easily enough to keep track of that state in the app, but the problem comes when you close the Modal. On close the modal removes the `open` attribute, but it doesn't indicate that to the app so the apps state is out of sync. Then when you click the toggle button you have to click it twice to firstly toggle the state back to false and secondly toggle the state back to true.

Because of this we need to flip the attribute not just based on state, but also based on whether the attribute is there or not. This brings about a question I had about Modal previously which is should Modal be removing the `open` attribute at all? I think it is okay still because we have a solution, but I am just wondering if there is a better way.

Reply:
so removing `open` is optional? We don't have to remove it?

Reply:
Removing `open` is not optional, hence the ternary that sets the attribute to `null` in order to remove it when the value of `isModalOpen` is `false`.